ray-operator/controllers/ray/batchscheduler/koordinator/koordinator_scheduler.go

```diff
@@ -0,0 +1,111 @@
+package koordinator
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	schedulerinterface "github.com/ray-project/kuberay/ray-operator/controllers/ray/batchscheduler/interface"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+)
+
+const (
+	SchedulerName                             string = "koordinator"
+	KoordinatorGangAnnotationName             string = "gang.scheduling.koordinator.sh/name"
+	KoordinatorGangMinAvailableAnnotationName string = "gang.scheduling.koordinator.sh/min-available"
+	KoordinatorGangTotalNumberAnnotationName  string = "gang.scheduling.koordinator.sh/total-number"
+	KoordinatorGangModeAnnotationName         string = "gang.scheduling.koordinator.sh/mode"
+	KoordinatorGangGroupsAnnotationName       string = "gang.scheduling.koordinator.sh/groups"
+
+	KoordinatorGangModeStrict string = "Strict"
+)
+
+type KoodinatorScheduler struct{}
+
+type KoordinatorSchedulerFactory struct{}
+
+func GetPluginName() string {
+	return SchedulerName
+}
+
+func (y *KoodinatorScheduler) Name() string {
+	return GetPluginName()
+}
+
+func (y *KoodinatorScheduler) DoBatchSchedulingOnSubmission(_ context.Context, _ *rayv1.RayCluster) error {
+	// koordinator doesn't require any resources to be created upfront
+	// this is a no-opt for this implementation
+	return nil
+}
+
+// AddMetadataToPod adds essential labels and annotations to the Ray pods
+// the koordinator scheduler needs these labels and annotations in order to do the scheduling properly
+func (y *KoodinatorScheduler) AddMetadataToPod(ctx context.Context, app *rayv1.RayCluster, groupName string, pod *corev1.Pod) {
+	logger := ctrl.LoggerFrom(ctx).WithName(SchedulerName)
+
+	// when gang scheduling is enabled, extra annotations need to be added to all pods
+	if y.isGangSchedulingEnabled(app) {
```

Comment:
```go
if !y.isGangSchedulingEnabled(app) {
    return
}

....
```

Reply:
> ```go
> if !y.isGangSchedulingEnabled(app) {
>     return
> }
> 
> ....
> ```

Thank you for your review! I will fix it as soon as possible.